src/components/panel/panel.spec.js

```diff
@@ -1049,6 +1049,169 @@ describe('$mdPanel', function() {
+
+      expect(getNumberOfGroups()).toEqual(1);
+
+      $mdPanel.newPanelGroup('test');
+
+      expect(getNumberOfGroups()).toEqual(1);
+    });
+
+    it('should not create a group using the config option when the group is ' +
+        'already defined', function() {
+          $mdPanel.newPanelGroup('test');
+
+          expect(getNumberOfGroups()).toEqual(1);
+
+          var config = {
+            groupName: 'test'
+          };
+          var panel = $mdPanel.create(config);
+
+          expect(getNumberOfGroups()).toEqual(1);
+        });
+
+    it('should add a panel to a group using the addToGroup method', function() {
+      $mdPanel.newPanelGroup('test');
+      var panel = $mdPanel.create(DEFAULT_CONFIG);
+
+      panel.addToGroup('test');
+      expect(getGroupPanels('test')).toContain(panel);
+    });
+
+    it('should add a panel to a group using the config option groupName',
+        function() {
+          $mdPanel.newPanelGroup('test');
+
+          var config = {
+            groupName: 'test'
+          };
+
+          var panel = $mdPanel.create(config);
+          expect(getGroupPanels('test')).toContain(panel);
+        });
+
+    it('should remove a panel from a group using the removeFromGroup method',
+        function() {
+          $mdPanel.newPanelGroup('test');
+
+          var config = {
+            groupName: 'test'
+          };
+
+          var panel = $mdPanel.create(config);
+
+          panel.removeFromGroup('test');
+          expect(getGroupPanels('test')).not.toContain(panel);
+        });
+
+    it('should remove a panel from a group on panel destroy', function() {
+      $mdPanel.newPanelGroup('test');
+
+      var config = {
+        groupName: 'test'
+      };
+
+      var panel = $mdPanel.create(config);
+
+      panel.destroy();
+      expect(getGroupPanels('test')).not.toContain(panel);
+    });
+
+    it('should set the maximum number of panels allowed open within a group ' +
+        'using the newPanelGroup option', function() {
+          $mdPanel.newPanelGroup('test', {
+            maxOpen: 1
+          });
+
+          expect(getGroupMaxOpen('test')).toEqual(1);
+        });
+
+    it('should set the maximum number of panels allowed open within a group ' +
+        'using the setGroupMaxOpen method', function() {
+          $mdPanel.newPanelGroup('test');
+          $mdPanel.setGroupMaxOpen('test', 1);
+
+          expect(getGroupMaxOpen('test')).toEqual(1);
+        });
+
+    it('should throw if trying to set maxOpen on a group that doesn\'t exist',
+        function() {
+          var expression = function() {
+            $mdPanel.setGroupMaxOpen('test', 1);
+          };
+
+          expect(expression).toThrow();
+        });
+
+    it('should update open panels when a panel is closed', function() {
+      $mdPanel.newPanelGroup('test');
+
+      var config = {
+        groupName: 'test'
+      };
+
+      var panel = $mdPanel.create(config);
+
+      panel.open();
```

Comment:
@ErinCoughlan Unless I am mistaken, this change will disrupt the expect tests that I am running in this test. I used this method because I needed to create a panel variable to then check the open panels array for afterward. Please let me know if I didn't write this test correctly.